### DISCUSSION TITLE
Runway v7 augmentable issue

### DISCUSSION THREAD

**Discussion by ivang76** [statamic-rad-pack/runway#493] [Help]
### Description

Hi there, I just updated one project to the latest Statamic and Runway versions and now I get this error on some of my models:

> Declaration of App\Models\Feed::augmented() must be compatible with Statamic\Contracts\Data\Augmentable::augmented(): Statamic\Contracts\Data\Augmented

The definition of the model is something like this, and worked without problem until the update:
```
[...]
use Statamic\Contracts\Query\ContainsQueryableValues;
use Statamic\Contracts\Data\Augmentable;

class Feed extends Model implements SearchableContract, ContainsQueryableValues, Augmentable
{
    use HasFactory;
    use HasRunwayResource; 
    use Searchable;

    public $timestamps = false;
    ...
    ...
    
 ```
 
 any ideas on how to fix it?
 thanks
 

### Environment

Symfony\Component\ErrorHandler\Error\FatalError 

  Class App\Models\Feed contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (Statamic\Contracts\Data\Augmentable::toDeferredAugmentedArray, Statamic\Contracts\Data\Augmentable::toDeferredAugmentedArrayUsingFields)

  at app/Models/Feed.php:20
     16▕ use Illuminate\Database\Eloquent\Builder;
     17▕ use Carbon\Carbon;
     18▕ use Statamic\Contracts\Data\Augmented;
     19▕ 
  ➜  20▕ class Feed extends Model implements SearchableContract, ContainsQueryableValues, Augmentable
     21▕ {
     22▕     use HasFactory;
     23▕     use HasRunwayResource; 
     24▕     use Searchable;

   Whoops\Exception\ErrorException 

  Class App\Models\Feed contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (Statamic\Contracts\Data\Augmentable::toDeferredAugmentedArray, Statamic\Contracts\Data\Augmentable::toDeferredAugmentedArrayUsingFields)

  at app/Models/Feed.php:20
     16▕ use Illuminate\Database\Eloquent\Builder;
     17▕ use Carbon\Carbon;
     18▕ use Statamic\Contracts\Data\Augmented;
     19▕ 
  ➜  20▕ class Feed extends Model implements SearchableContract, ContainsQueryableValues, Augmentable
     21▕ {
     22▕     use HasFactory;
     23▕     use HasRunwayResource; 
     24▕     use Searchable;

**Comment by duncanmcclean** [comment]
Out of interest, why are you handling search yourself rather than using [Runway's search feature](https://runway.duncanmcclean.com/search)?

The exceptions don't seem to have anything to do with Runway, but rather some changes you need to make to how you're handling augmentation on your model for your own search stuff.

**Comment by ivang76** [comment]
I'm testing both the ways for internal development and some customisations. Is there some documentation about those changes I should need to do for the upgrade? Everything worked without problem before, and I can't stay with Runway 6 if I want to go with Statamic 5... thanks

**Reply by duncanmcclean** [reply]
There's no documentation around it, really. Most people aren't implementing augmentation themselves. 

Statamic 5 re-worked augmentation in various ways to make it more performant. You might be better opening a discussion on the `statamic/cms` repository, since it's not Runway related.
